include/gbt.h

```diff
@@ -11,96 +11,67 @@
- *  - `string error` : If an error occurs, this keys exists and contains the error message.
- */
-void scanGBTPhases(const RPCMsg *request, RPCMsg *response);
-
-/*! \brief Local callable version of `scanGBTPhase`.
- *  \param[in, out] la Local arguments structure.
  *  \param[in] ohN OptoHybrid index number.
  *  \param[in] N The number of times the scan must performed.
  *  \param[in] phaseMin Lowest phase to scan (min = 0).
  *  \param[in] phaseMax Highest phase to scan (max = 15).
  *  \param[in] phaseStep Step to scan the phases.
- *  \return Returns `false` in case of success; `true` in case of error. The precise error is logged and written to the `error` RPC key.
+ *  \return Returns a 2-dimensional array containing each phase of each VFAT in each OptoHybrid. The precise error is logged and written to the `LOG4CPLUS` logger.
  *
  *  \detail
  *
  *  The scan seeks for valid RX phases for the VFAT's of one OptoHybrid. A phase is considered valid when `LINK_GOOD = 1`, `SYNC_ERR_CNT = 0` and `CFG_RUN != 0xdeaddead`. In order to improve the reliability of the scan, it is repeated `N` times.
  *
- *  The results are returned as RPC keys named `OHX.VFATY` where X is the index of the OptoHyrid and Y the index of the VFAT within this OptoHybrid.
+ *  The results are returned as a 2-dimensional array in ``std::vector<std::vector<uint32_t>>`` format, where there is one vector entry per OptoHybrid index, and each entry of the sub-vectors corresponds to the phase read at that VFAT index within the OptoHybrid.
  *  Each key is a word array of 16 elements for the 16 possible phases ordered from phase 0 to phase 15.
  *  Each word is the number of time the scan was "good" out of the total number of scan requested, N.
  *
  */
-bool scanGBTPhasesLocal(localArgs *la, const uint32_t ohN, const uint32_t N = 1, const uint8_t phaseMin = gbt::PHASE_MIN, const uint8_t phaseMax = gbt::PHASE_MAX, const uint8_t phaseStep = 1);
+struct scanGBTPhases : public xhal::rpc::Method
 
-/*! \brief Write the GBT configuration of one OptoHybrid.
- *  \param[in] request RPC response message.
- *  \param[out] response RPC response message.
- *
- *  The method expects the following RPC keys :
- *  - `word ohN` : OptoHybrid index number.
- *  - `word gbtN` : Index of the GBT to configure. There 3 GBT's per OptoHybrid in the GE1/1 chambers.
- *  - `binarydata config` : Configuration blob of the GBT. This is a 366 element long vector of the registers to write. The registers are sorted from address 0 to address 365.
- *
- *  The method returns the following RPC keys :
- *  - `string error` : If an error occurs, this keys exists and contains the error message.
- */
-void writeGBTConfig(const RPCMsg *request, RPCMsg *response);
+{
+
+    std::vector<std::vector<uint32_t>> operator()(const uint32_t &ohN, const uint32_t &N = 1, const uint8_t &phaseMin = gbt::PHASE_MIN, const uint8_t &phaseMax = gbt::PHASE_MAX, const uint8_t &phaseStep = 1) const;
```

Comment:
while we're updating, we should also fix the argument names. single letter parameter names should not be used